tests/robustness/multiclient/client/client.go

```diff
@@ -0,0 +1,70 @@
+// +build darwin,amd64 linux,amd64
+
+// Package client manages client specific info
+package client
+
+import (
+	"context"
+	"sync"
+
+	petname "github.com/dustinkirkland/golang-petname"
+	"github.com/google/uuid"
+)
+
+type key int
+
+const (
+	clientKey key = 0
+	nameLen   int = 2
+)
+
+// Client is a unique client for use in multiclient robustness tests.
+type Client struct {
+	ID string
+}
+
+func newClient() *Client {
+	petname.NonDeterministicMode()
+
+	return &Client{
+		ID: petname.Generate(nameLen, "-") + "-" + uuid.NewString(),
+	}
+}
+
+// WrapContext returns a copy of ctx with a new client.
+func WrapContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, clientKey, newClient())
+}
```

Comment:
`int` with value `0` is a terrible key.

Suggestion:
```
var clientKey = struct{}{}
```

Reply:
I know it seems weird, but I actually did some reading to come up with that. On the Go blog, they recommend this as a way to avoid key name collisions. See section "Package userip" in https://blog.golang.org/context

Reply:
And here is a rebuttal to the very same blog: https://medium.com/random-go-tips/on-context-context-keys-uniqueness-738aa7a5c87f